sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -203,22 +206,35 @@ export class ECertFileHandler extends ESDCFileHandler {
    * generate the record.
    * @returns e-Cert record.
    */
-  private createECertRecord(disbursement: DisbursementSchedule): ECertRecord {
+  private createECertRecord(
+    disbursement: DisbursementScheduleWithStopFullTimeBCFundingStatus,
+    offeringIntensity: OfferingIntensity,
+  ): ECertRecord {
     const now = new Date();
     const application = disbursement.studentAssessment.application;
     const addressInfo = application.student.contactInfo.address;
     const offering = application.currentAssessment.offering;
     const fieldOfStudy = getFieldOfStudyFromCIPCode(
       offering.educationProgram.cipCode,
     );
-    const awards = disbursement.disbursementValues.map(
-      (disbursementValue) =>
-        ({
-          valueType: disbursementValue.valueType,
-          valueCode: disbursementValue.valueCode,
-          valueAmount: disbursementValue.valueAmount,
-        } as Award),
-    );
+    const awards = [];
+    for (const disbursementValue of disbursement.disbursementValues) {
+      if (
+        disbursement.stopFullTimeBCFundingStatus &&
```

Comment:
We have a file to process part-time and another to process full-time. This feature is not a concern at all for part-time hence it must be added only to the full-time processing on file `e-cert-full-time-integration\e-cert-full-time-integration.service.ts`.

Reply:
I also initially thought about it, but the reason for creating the complete code for a simple if condition, creates code duplication. That is the reason to have it. I have done the code change, please confirm if we can ignore the code duplication. Thanks